src/utils/observability/src/panic_handler.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::any::Any;
+use std::panic;
+use std::sync::{Arc, Mutex};
+
+use axum::body::Body;
+use http::{header, Response, StatusCode};
+use lazy_static::lazy_static;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Such as we can't get real backtrace in catch_unwind block
+// we need place to store trace if such occurs and then print it in logs
+lazy_static! {
+    static ref BACKTRACE: Arc<Mutex<Option<String>>> = Arc::new(Mutex::new(None));
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub fn set_panic_hook() {
+    panic::set_hook(Box::new(move |_| {
+        *BACKTRACE.lock().unwrap() = Some(std::backtrace::Backtrace::force_capture().to_string());
+    }));
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[allow(clippy::needless_pass_by_value)]
+pub fn panic_handler(err: Box<dyn Any + Send + 'static>) -> Response<Body> {
+    let panic_msg = if let Some(s) = err.downcast_ref::<&str>() {
+        *s
+    } else if let Some(s) = err.downcast_ref::<String>() {
+        s.as_str()
+    } else {
+        "Unknown panic payload"
+    };
+
+    let backtrace = BACKTRACE
+        .lock()
+        .unwrap()
+        .take()
+        .unwrap_or("Backtrace not found".to_string());
+
+    tracing::error!(err = "Unhandled panic caught", panic_msg, %backtrace);
```

Comment:
The way this is logged is different from how we log other things so (given how sensitive and error prone `tracing` macros are) this makes me question how it looks in the log.

Recommend changing this to the style of all our other tracing events:
```rust
tracing::errror!(panic_msg, %backtrace, "Unhandled panic caught");
```